AdyenNetworking/APIClient/APIClient.swift

```diff
@@ -136,10 +136,14 @@ public final class APIClient: APIClientProtocol {
                         }
                     }
                 
-                completionHandler(result)
+                DispatchQueue.main.sync {
+                    completionHandler(result)
+                }
             }.resume()
         } catch {
-            completionHandler(.failure(error))
+            DispatchQueue.main.sync {
```

Comment:
i don't think `DispatchQueue.main.sync` is a good idea, its a bad practice in general because it can easily lead to deadlocks :-).